src/mlpack/methods/hmm/hmm_impl.hpp

```diff
@@ -522,6 +522,80 @@ double HMM<Distribution>::LogLikelihood(const arma::mat& dataSeq) const
   return accu(logScales);
 }
 
+/**
+ * Compute the log of the scaling factor of the given emission probability
+ * at time t. To calculate the log-likelihood for the whole sequence,
+ * accumulate log scale over the entire sequence
+ */
+template<typename Distribution>
+double HMM<Distribution>::EmissionLogScaleFactor(
+    const arma::vec& emissionLogProb,
+    arma::vec& forwardLogProb) const
+{
+    double curLogScale;
```

Comment:
```suggestion
  double curLogScale;
```

I can't do multiline suggestions on a phone; I'm really sorry for all the tiny suggestions. :(